specs/governance/alligator.md

```diff
@@ -0,0 +1,179 @@
+transfer. This enables the `Alligator` contract to consume the hook and update its delegation and checkpoint mappings
+accordingly. If either address involved in the transfer (`_from_` or `_to`) has not been migrated to the `Alligator` contract,
+the contract copies the address' delegation and checkpoint data from the token contract to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address with specified subdelegation rules. This function
+is inteded to be called by users.
+
+```solidity
+subdelegate(address _token, address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address with specified subdelegation rules. This function
+is intended to be called by the token contract.
+
+```solidity
+subdelegate(address _account, address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+subdelegateBatched(address _token, address[] _delegatees, SubdelegationRule[] _rules)
+```
+
+Calls the `subdelegate` function for each pair of `_delegatees` address and subdelegation rule.
+
+#### `afterTokenTransfer`
+
+Updates delegation and checkpoint mappings for the address of a token contract after a transfer. This function
+MUST be called by the `_afterTokenTransfer` function in the token contract.
+
+```solidity
+afterTokenTransfer(address _from, address _to, uint256 _amount)
+```
+
+If the `_to` or `_from` addresses have not been migrated, `Alligator` migrates by copying the delegation and checkpoint
+data from the token contract to its own state.
+
+### Getters
+
+The output for these functions is conditional on whether the user address has been migrated or not. Concretely, the
+`Alligator` MUST use its own state if the address has been migrated, or else it MUST use the state of the token contract.
+
+#### `getSubdelegations`
+
+Retrieves the subdelegations for a given token contract and user address.
+
+```solidity
+getSubdelegations(address _token, address _user) returns (SubdelegationRule[] memory)
+```
+
+#### `getCheckpoints`
+
+Retrieves the checkpoints for a given token contract and user address.
+
+```solidity
+getCheckpoints(address _token, address _user) returns (Checkpoint[] memory)
+```
+
+#### `getVotingPower`
+
+Retrieves the current and past voting power for a given token contract and user address.
+
+```solidity
+getVotingPower(address _token, address _user, uint256 _blockNumber) returns (uint256)
+```
+
+## Storage
+
+The `Alligator` contract MUST be able to store delegations and checkpoints for multiple token contracts.
+These storage variables MUST be defined in the same way as in the token contract:
+
```

Comment:
Its a bit easier to read when the descriptions are outside of the code block, see https://github.com/ethereum-optimism/specs/pull/247#discussion_r1681672402

Reply:
yup, updated with same style as in https://github.com/ethereum-optimism/specs/blob/main/specs/interop/messaging.md#message-identifier